volume/volume.go

```diff
@@ -73,6 +73,12 @@ type LabeledVolume interface {
 	Volume
 }
 
+// OptionedVolume wraps a Volume with driver defined options
+type OptionedVolume interface {
```

Comment:
@cpuguy83 The `VolumeWithDetails` has been changed to `DetailedVolume` as otherwise golint will complain.